src/Elliptic/Systems/Elasticity/BoundaryConditions/LaserBeam.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Elasticity/BoundaryConditions/LaserBeam.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Elasticity::BoundaryConditions::detail {
+
+void LaserBeamImpl::apply(
+    const gsl::not_null<tnsr::I<DataVector, 3>*> /*displacement*/,
+    const gsl::not_null<tnsr::I<DataVector, 3>*> n_dot_minus_stress,
+    const tnsr::I<DataVector, 3>& x,
+    const tnsr::i<DataVector, 3>& face_normal) const noexcept {
+  const auto n_dot_x = dot_product(face_normal, x);
+  const auto r_sq = square(get<0>(x)) + square(get<1>(x)) + square(get<2>(x)) -
```

Comment:
Here you appear to specifically not use the `magnitude` function to (presumably) speed things up by avoiding allocations and getting an expression template instead, but you're not doing the same thing for `n_dot_x` -- is there a reason to treat them differently?

Reply:
no reason, switched to the expression template as well